src/components/carEditDialog.tsx

```diff
@@ -0,0 +1,101 @@
+    const [errors, setErrors] = useState<FormErrors>(emptyCar)
+
+    useEffect(() => {
+        setCar(carData === undefined ? emptyCar : carData)
+    }, [carData])
+
+    useEffect(() => { // Effect to re-validate form fields
+        if (Object.values(errors).some((value) => value !== '')){
+            validateCar({car, errors, setErrors})
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [car])
+
+    const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+        const { name, value } = e.target
+        setCar({ ...car, [name]: value })
+    }
+
+    const handleSubmit = () => {
+        if (!validateCar({car, errors, setErrors})) return
+        editCar(car)
+        handleClose()
+    }
+
+    return (
+        <Dialog open={open} onClose={handleClose}>
+            <DialogTitle>Edit Car</DialogTitle>
+            <DialogContent>
+                <TextField
+                    sx={{marginY: 1}}
+                    label='Maker'
+                    name='maker'
+                    select
+                    value={car.maker}
+                    onChange={handleChange}
+                    error={errors.maker !== ''}
+                    helperText={errors.maker}
+                    fullWidth>
+                        <MenuItem value="">Select Maker</MenuItem>
+                        <MenuItem value="Toyota">Toyota</MenuItem>
+                        <MenuItem value="Honda">Honda</MenuItem>
+                        <MenuItem value="Ford">Ford</MenuItem>
+                </TextField>
+                <TextField
+                    sx={{marginY: 1}}
+                    label='Model'
+                    name='model'
+                    value={car.model}
+                    onChange={handleChange}
+                    error={errors.model !== ''}
+                    helperText={errors.model}
+                    fullWidth
+                />
+                <TextField
+                    sx={{marginY: 1}}
+                    label='Year'
+                    name='year'
+                    value={car.year}
+                    onChange={handleChange}
+                    error={errors.year !== ''}
+                    helperText={errors.year}
+                    fullWidth
+                />
+                <FormControl required component="fieldset" error={Boolean(errors.engine)}>
+                    <FormLabel component="legend">Engine Type</FormLabel>
+                    <RadioGroup 
+                        row
+                        aria-label="engine" 
+                        name="engine" 
+                        value={car.engine} 
+                        onChange={handleChange}>
+                        <FormControlLabel value="Petrol" control={<Radio />} label="Petrol" />
+                        <FormControlLabel value="Diesel" control={<Radio />} label="Diesel" />
+                        <FormControlLabel value="Electric" control={<Radio />} label="Electric" />
```

Comment:
Instead of duplicating the component for button for each engine, you can create an `enum` or `array` of engines and then render button for each of element using `map`:
`engines.map((engine) =>  <FormControlLabel value={engine} control={<Radio />} label={engine} />);`

That's will be much convenient way if you decide to add some styles to buttons, since you don't need to duplicate it for each option